volcli/volcli.go

```diff
@@ -286,6 +287,115 @@ func policyList(ctx *cli.Context) (bool, error) {
 	return false, nil
 }
 
+// PolicyGetRevision retrieves a single revision from a policy's history.
+func PolicyGetRevision(ctx *cli.Context) {
+	execCliAndExit(ctx, policyGetRevision)
+}
+
+func policyGetRevision(ctx *cli.Context) (bool, error) {
+	if len(ctx.Args()) != 2 {
+		return true, errorInvalidArgCount(len(ctx.Args()), 2, ctx.Args())
+	}
+
+	name := ctx.Args()[0]
+	revision := ctx.Args()[1]
+
+	resp, err := http.Get(fmt.Sprintf("http://%s/policy-archives/%s/%s",
+		ctx.GlobalString("apiserver"),
+		name,
+		revision,
+	))
+	if err != nil {
+		return false, err
+	}
+
+	if resp.StatusCode != 200 {
+		if _, err := io.Copy(os.Stderr, resp.Body); err != nil {
+			return false, errored.Errorf("Error copying body: %v\nResponse Status Code was %d, not 200", err, resp.StatusCode)
+		}
+		return false, errored.Errorf("Response Status Code was %d, not 200", resp.StatusCode)
+	}
+
+	content, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, err
+	}
+
+	fmt.Println(string(content))
+
+	return false, nil
+}
+
+// PolicyListRevisions retrieves all the revisions for a given policy.
+func PolicyListRevisions(ctx *cli.Context) {
+	execCliAndExit(ctx, policyListRevisions)
+}
+
+func policyListRevisions(ctx *cli.Context) (bool, error) {
+	var revisions []string
+
+	if len(ctx.Args()) != 1 {
+		return true, errorInvalidArgCount(len(ctx.Args()), 1, ctx.Args())
+	}
+
+	name := ctx.Args()[0]
+
+	resp, err := http.Get(fmt.Sprintf("http://%s/policy-archives/%s",
+		ctx.GlobalString("apiserver"),
+		name,
+	))
+	if err != nil {
+		return false, err
+	}
+
+	if resp.StatusCode != 200 {
+		if _, err := io.Copy(os.Stderr, resp.Body); err != nil {
+			return false, errored.Errorf("Error copying body: %v\nResponse Status Code was %d, not 200", err, resp.StatusCode)
+		}
+		return false, errored.Errorf("Response Status Code was %d, not 200", resp.StatusCode)
+	}
+
+	content, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return false, err
+	}
+
+	if err := json.Unmarshal(content, &revisions); err != nil {
+		return false, err
+	}
+
+	for _, revision := range revisions {
+		fmt.Println(revision)
+	}
+
+	return false, nil
+}
+
+// PolicyWatch watches etcd for policy changes and prints them.
+func PolicyWatch(ctx *cli.Context) {
+	execCliAndExit(ctx, policyWatch)
+}
+
+func policyWatch(ctx *cli.Context) (bool, error) {
+	if len(ctx.Args()) != 0 {
+		return true, errorInvalidArgCount(len(ctx.Args()), 0, ctx.Args())
+	}
+
+	cfg, err := config.NewClient(ctx.GlobalString("prefix"), ctx.GlobalStringSlice("etcd"))
+	if err != nil {
+		return false, err
+	}
+
+	c := make(chan *watch.Watch)
+
+	cfg.WatchForPolicyChanges(c)
+
+	for range c {
+	}
```

Comment:
Originally, no, but based on my other comment, I will change it so that the watcher just emits the data and lets the establisher of the watch decide what to do with it (in this case, print it to stdout).